api/api.proto

```diff
@@ -944,6 +973,7 @@ message Volume {
   bool in_trashcan = 31;
   // DerivedIoProfile the IO profile determined from the pattern
   IoProfile derived_io_profile = 32;
+  // TODO should we add sharedv4_spec here?
```

Comment:
It is better to add it here. If an input spec is not provided we can set the default value in the Volume object.

Reply:
In that case, it makes sense to not add it because the default depends on the type of volume (sharedv4 vs sharedv4 service). If the value is not specified, we use the appropriate default.

Reply:
So shouldn't we be setting that value here? Essentially when someone inspects the volume they will know what strategy is being used by looking under this Volume struct.

Reply:
I was thinking that we want to leave it empty to indicate that the user does not have a preference and wants to use the appropriate system default. If the volume type is changed from sharedv4 -> sharedv4 service, a different default value is used automatically.